cmd/debug.go

```diff
@@ -0,0 +1,184 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/lf-edge/eden/pkg/defaults"
+	"github.com/lf-edge/eden/pkg/utils"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var perfOptions string
+
+var debugCmd = &cobra.Command{
+	Use: "debug",
+}
+
+var debugStartEveCmd = &cobra.Command{
+	Use:   "start",
+	Short: "start perf in EVE",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		assignCobraToViper(cmd)
+		viperLoaded, err := utils.LoadConfigFile(configFile)
+		if err != nil {
+			return fmt.Errorf("error reading config: %s", err.Error())
+		}
+		if viperLoaded {
+			eveSSHKey = utils.ResolveAbsPath(viper.GetString("eden.ssh-key"))
+			extension := filepath.Ext(eveSSHKey)
+			eveSSHKey = strings.TrimRight(eveSSHKey, extension)
+			eveRemote = viper.GetBool("eve.remote")
+			eveRemoteAddr = viper.GetString("eve.remote-addr")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		eveIP := getEVEIP()
+		if eveIP == "" {
+			log.Fatal("Np EVE IP")
+		}
+		if eveRemote || eveRemoteAddr == "" {
+			if !cmd.Flags().Changed("eve-ssh-port") {
+				eveSSHPort = 22
+			}
+		}
+		if _, err := os.Stat(eveSSHKey); !os.IsNotExist(err) {
+			commandToRun := fmt.Sprintf("perf record %s -o %s", perfOptions, defaults.DefaultPerfEVELocation)
+			arguments := fmt.Sprintf("-o ConnectTimeout=5 -oStrictHostKeyChecking=no -i %s -p %d root@%s %s", eveSSHKey, eveSSHPort, eveIP, commandToRun)
+			log.Debugf("Try to ssh %s:%d with key %s and command %s", eveHost, eveSSHPort, eveSSHKey, arguments)
+			if _, err := utils.RunCommandBackground("ssh", nil, strings.Fields(arguments)...); err != nil {
+				log.Fatalf("ssh error for command %s: %s", commandToRun, err)
+			}
+		} else {
+			log.Fatalf("SSH key problem: %s", err)
+		}
+	},
+}
+
+var debugStopEveCmd = &cobra.Command{
+	Use:   "stop",
+	Short: "stop perf in EVE",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		assignCobraToViper(cmd)
+		viperLoaded, err := utils.LoadConfigFile(configFile)
+		if err != nil {
+			return fmt.Errorf("error reading config: %s", err.Error())
+		}
+		if viperLoaded {
+			eveSSHKey = utils.ResolveAbsPath(viper.GetString("eden.ssh-key"))
+			extension := filepath.Ext(eveSSHKey)
+			eveSSHKey = strings.TrimRight(eveSSHKey, extension)
+			eveRemote = viper.GetBool("eve.remote")
+			eveRemoteAddr = viper.GetString("eve.remote-addr")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
```

Comment:
This thing does the same as eden eve ssh AFAIU 
Maybe reuse eden eve ssh functions? Or use this there?

Reply:
Actually, in debug.go we run commands in foreground, in background, scp/ssh, docker run... I do not thing that we can reuse eve ssh efficiently.